### DISCUSSION TITLE
Converting model checkpoint(.pth) to HuggingFace Model

### DISCUSSION THREAD

**Discussion by sunilswain** [h2oai/h2o-llmstudio#730] [Q&A] [answered]
Hey
I've tried finetuinng my model using the llmstudio cli tool. And finally I have the experiments files now, I'm curious to know how can push my model to huggingface.
Here is the list of my files
![image](https://github.com/h2oai/h2o-llmstudio/assets/69412831/db093bc3-176a-45cd-b2c1-4b8d1967bbcf)

Thanks

**Comment by pascal-pfeiffer** [comment, marked as answer]
When using the UI, you can follow the steps here (https://h2oai.github.io/h2o-llmstudio/guide/experiments/export-trained-model) to push your model to Hugging Face Hub or to download locally. 

When using the CLI, use the [publish_to_huggingface.py](https://github.com/h2oai/h2o-llmstudio/blob/main/publish_to_hugging_face.py) script.
```bash
make shell
python publish_to_huggingface.py --path_to_experiment YOUR_PATH 
```
There are a few more args to fill if you need them. Show with:
```bash
python publish_to_huggingface.py --help
```
